### PR TITLE
[Button] Reduce code complexity

### LINKED ISSUE #10983: [Button] Variant / Tone checks are overly complex
Button has variant and tone checks that look like this:

```ts
variant === 'primary' && styles.primary,
variant === 'plain' && styles.plain,
variant === 'tertiary' && styles.primary,
variant === 'tertiary' && styles.tertiary,
variant === 'monochromePlain' && styles.monochrome,
variant === 'monochromePlain' && styles.plain,
tone === 'critical' && styles.critical,
tone === 'success' && styles.success,
```

There are multiple checks for variants that apply different classes leading to code that's harder to maintain. We should be able to simplify this by using the `variationName` helper and cleaning up some css

### PR DESCRIPTION
**Pull request by kyledurand**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris/issues/10983

### WHAT is this pull request doing?

Opting to use the `variationName` helper to cleanup how class names get set and remove some nested css

[🌀 Spin](https://admin.web.web-g78t.kyle-durand.us.spin.dev/store/shop1)

### CODE REVIEW

File: polaris-react/src/components/Button/Button.scss [hunk 1]
```diff
@@ -768,9 +765,7 @@
   }
 
   &.sizeLarge {
-    // stylelint-disable-next-line polaris/conventions/polaris/custom-property-allowed-list -- custom property
-    margin: calc(-1 * var(--pc-button-large-vertical-padding))
-      calc(-1 * var(--p-space-500));
+    margin: calc(-1 * var(var(--p-space-200))) calc(-1 * var(--p-space-500));
```
Comment (aveline) [inline polaris-react/src/components/Button/Button.scss]: This should also be split out into `margin-block` and `margin-inline` for consistency

File: polaris-react/src/components/Button/Button.scss [hunk 2]
```diff
@@ -919,11 +912,11 @@
...
 
 .sizeLarge {
-  // stylelint-disable polaris/conventions/polaris/custom-property-allowed-list -- Button custom properties
+  /* stylelint-disable-next-line polaris/conventions/polaris/custom-property-allowed-list -- custom height */
   min-height: var(--pc-button-large-min-height);
+  /* stylelint-disable-next-line polaris/conventions/polaris/custom-property-allowed-list -- custom height */
   min-width: var(--pc-button-large-min-height);
-  padding: var(--pc-button-large-vertical-padding) var(--p-space-300);
-  // stylelint-enable polaris/conventions/polaris/custom-property-allowed-list
+  padding: var(var(--p-space-200)) var(--p-space-300);
```
Comment (aveline) [inline polaris-react/src/components/Button/Button.scss]: Split into `inline` and `block` for consistency with `margin`?

**Review by aveline — APPROVED** [review]

File: polaris-react/src/components/Button/Button.scss [hunk 3]
```diff
@@ -768,9 +766,8 @@
   }
 
   &.sizeLarge {
-    // stylelint-disable-next-line polaris/conventions/polaris/custom-property-allowed-list -- custom property
-    margin: calc(-1 * var(--pc-button-large-vertical-padding))
-      calc(-1 * var(--p-space-500));
+    margin-block: calc(-1 * var(var(--p-space-200)));
```
Comment (laurkim) [inline polaris-react/src/components/Button/Button.scss]:
```suggestion
    margin-block: calc(-1 * var(--p-space-200));
```

File: polaris-react/src/components/Button/Button.scss [hunk 4]
```diff
@@ -849,10 +846,8 @@
   }
 
   &.sizeLarge {
-    // stylelint-disable polaris/conventions/polaris/custom-property-allowed-list -- custom properties
-    padding-left: var(--pc-button-large-vertical-padding);
-    padding-right: var(--pc-button-large-vertical-padding);
-    // stylelint-enable polaris/conventions/polaris/custom-property-allowed-list -- custom properties
+    padding-left: var(var(--p-space-200));
+    padding-right: var(var(--p-space-200));
```
Comment (laurkim) [inline polaris-react/src/components/Button/Button.scss]:
```suggestion
    padding-left: var(--p-space-200);
    padding-right: var(--p-space-200);
```

File: polaris-react/src/components/Button/Button.scss [hunk 5]
```diff
@@ -919,11 +914,12 @@
...
 
 .sizeLarge {
-  // stylelint-disable polaris/conventions/polaris/custom-property-allowed-list -- Button custom properties
+  /* stylelint-disable-next-line polaris/conventions/polaris/custom-property-allowed-list -- custom height */
   min-height: var(--pc-button-large-min-height);
+  /* stylelint-disable-next-line polaris/conventions/polaris/custom-property-allowed-list -- custom height */
   min-width: var(--pc-button-large-min-height);
-  padding: var(--pc-button-large-vertical-padding) var(--p-space-300);
-  // stylelint-enable polaris/conventions/polaris/custom-property-allowed-list
+  padding-block: var(var(--p-space-200));
```
Comment (laurkim) [inline polaris-react/src/components/Button/Button.scss]:
```suggestion
  padding-block: var(--p-space-200);
```

**Review by laurkim — APPROVED** [review]
Nice cleanup!! 🌟 💯 🙌 

Some minor feedback and [this line](https://github.com/Shopify/polaris/blob/11ded8195aad5296ef1ba8b5279b5c9cad2aa3f3/polaris-react/src/components/Button/Button.scss#L1000) doesn't show up in the diff so I couldn't add a suggestion but just need to update the `.primary` to `.variantPrimary` I think.

**Pull request merged**